estate/models/estate_property.py

```diff
@@ -0,0 +1,84 @@
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    
+    name = fields.Char("Property Name", required=True)
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Available From", copy=False, default=fields.Date.today() + relativedelta(months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    active = fields.Boolean(default=True)
+    garden_orientation = fields.Selection( 
+        string="Garden Orientation",
+        selection=[
+            ('north', 'North'), 
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ],
+    )
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True, copy=False, default='new',
+    )
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    user_id = fields.Many2one('res.users', string="Salesman", default=lambda self: self.env.user)
+    partner_id = fields.Many2one('res.partner', string="Buyer", copy=False)
+    tag_ids=fields.Many2many('estate.property.tag', string="Tags")
+    offer_ids=fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    total_area = fields.Integer(string='Total Area (sqm)', compute='_compute_total_area')
+    best_price = fields.Float(string='Best Offer', compute='_compute_best_offer')
+
+    @api.depends('living_area','garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area+record.garden_area
+            
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_price = max(record.offer_ids.mapped('price'))
+            else:
+                record.best_price = 0
+    
+    @api.onchange('garden')
+    def _onchange_garden(self): 
+        if self.garden == True:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False  
+    
+    def property_sold_action(self):
+        for record in self:
+            if record.state == 'cancelled':
+                raise UserError("Cancelled properties cannot be sold.")
```

Comment:
```suggestion
                raise UserError("Cancelled properties cannot be sold.")
```
what is the diff b/w `UserError` and `ValidationError`

Reply:
UserError is used to restrict user actions (restricts workflow).
ValidationError is used to validate database fields/data integrity and prevent records to saved.

Reply:
resolve it